### PR TITLE
Convert ShieldOutlineIcon to function Component

### LINKED ISSUE #24756: Convert ./components/widgets/icons/shield_outline_icon.tsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-54792

### PR DESCRIPTION
**Pull request by dipandhali2021**
#### Summary
This PR convert shieldOutlineComponet from class to function component

#### Ticket Link
Fixes #24756
Jira MM-54792
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/widgets/icons/shield_outline_icon.tsx [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
+
+const ShieldOutlineIcon = (props: Props) => {
+    return (
+        <span {...props}>
+            <svg
+                width='100%'
+                height='100%'
+                viewBox='0 0 24 24'
+            >
+
```
Comment (jespino) [inline webapp/channels/src/components/widgets/icons/shield_outline_icon.tsx:16]: nit: unneeded empty line

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @dipandhali2021! :tada:

I'm adding a comment about an empty line, feel free to fix that, but it is not mandatory to get this merged.

**Comment by dipandhali2021** [comment]
Hey @jespino , can you please guide me how to remove do-not-merge/release-note-label-needed label from the PR

**Comment by dipandhali2021** [comment]
Hey, @jespino . I have commited the suggested  changes and fixed the test file.

**Comment by jespino** [comment]
Nice! Thanks @dipandhali2021! :star_struck:

**Review by larkox — APPROVED** [review]
Great work!

**Review by lindy65 — APPROVED** [review]
Thanks @dipandhali2021 - looks good to me 👍 

cc @larkox

**Comment by larkox** [comment]
Thanks @dipandhali2021 for your contribution :tada:

**Pull request merged**